### PR TITLE
Make note of Ord's derive ordering

### LINKED ISSUE #26620: Document the implementation of #[derive(Ord)]
`(0, (1,0)) < (1, (0,1))` but `((1,0), 0) > ((0,1), 1)`
The order of the fields is important, and this is something that should be clearly documented, as changing the order of the fields of a struct can change program logic.

### PR DESCRIPTION
**Pull request by steveklabnik**
Fixes #26620

### CODE REVIEW

File: src/libcore/cmp.rs [hunk 1]
```diff
@@ -166,6 +166,8 @@ impl Ordering {
 ///
 /// - total and antisymmetric: exactly one of `a < b`, `a == b` or `a > b` is true; and
 /// - transitive, `a < b` and `b < c` implies `a < c`. The same must hold for both `==` and `>`.
+///
+/// When this trait is `derive`d, it produces a lexographic ordering.
```
Comment (alexcrichton) [inline src/libcore/cmp.rs]: lexicographic?

**Comment by alexcrichton** [comment]
r=me

File: src/libcore/cmp.rs (hunk 1 above)
Comment (steveklabnik) [inline src/libcore/cmp.rs]: i am the worst speller that ever spelled

**Pull request merged**